pkg/gce-pd-csi-driver/controller.go

```diff
@@ -716,6 +750,95 @@ func (gceCS *GCEControllerServer) createSingleDisk(ctx context.Context, req *csi
 	return disk, nil
 }
 
+func (gceCS *GCEControllerServer) diskSupportsIopsChange(diskType string) bool {
+	for _, disk := range gceCS.provisionableDisksConfig.SupportsIopsChange {
+		if disk == diskType {
+			return true
+		}
+	}
+	return false
+}
+
+func (gceCS *GCEControllerServer) diskSupportsThroughputChange(diskType string) bool {
+	for _, disk := range gceCS.provisionableDisksConfig.SupportsThroughputChange {
+		if disk == diskType {
+			return true
+		}
+	}
+	return false
+}
+
+func (gceCS *GCEControllerServer) ControllerModifyVolume(ctx context.Context, req *csi.ControllerModifyVolumeRequest) (*csi.ControllerModifyVolumeResponse, error) {
+	var err error
+
+	volumeID := req.GetVolumeId()
+	klog.V(4).Infof("Modifying Volume ID: %s", volumeID)
+
+	if volumeID == "" {
+		return nil, status.Error(codes.InvalidArgument, "volume ID must be provided")
+	}
+
+	diskType := metrics.DefaultDiskTypeForMetric
+	enableConfidentialCompute := metrics.DefaultEnableConfidentialCompute
+	enableStoragePools := metrics.DefaultEnableStoragePools
+
+	defer func() {
+		gceCS.Metrics.RecordOperationErrorMetrics("ModifyVolume", err, diskType, enableConfidentialCompute, enableStoragePools)
```

Comment:
The GRPC error code of this error is how we report errors to our SLO. Since it is in a defer statement, you must reassign err before returning in order for the correct error code to be sent to this function. For example, below, 
InvalidArgument will not get passed to RecordOperationErrorMetrics because you didn't reassign the err. Instead, it will pass whatever error code that was returned from VolumeIDToKey(volumeID).

```
	project, volKey, err := common.VolumeIDToKey(volumeID)
	if err != nil {
		return nil, status.Errorf(codes.InvalidArgument, "volume ID is invalid: %v", err.Error())
	}
```

Reply:
Done